R/finalize_plot.R

```diff
@@ -316,22 +325,35 @@ finalize_plot <- function(plot = NULL,
 #' Sub-fn to create plot grob, including legend-realignment
 #' @noRd
 prepare_chart <- function(plot,
-                         consts,
-                         overrides,
-                         legend_shift,
-                         debug,
-                         ...) {
+                          title,
+                          caption,
+                          consts,
+                          overrides,
+                          legend_shift,
+                          title_width,
+                          debug,
+                          ...) {
 
   # preformat plot ---------------------------------------------
 
   # the basics
-  plot <- plot + ggplot2::theme(
-    # remove any in-plot titles
-    plot.title = element_blank(),
-    plot.caption = element_blank(),
-    # apply any extra `ggplot2::theme()` args
-    ...
-  )
+  plot$labels$title <- title  # Embed supplied title in plot
+  plot$labels$caption <- caption  # Embed supplied caption in plot
+  if (title_width > 0 | caption == "") {
+    plot <- plot + ggplot2::theme(
+      plot.title = element_blank(),  # Remove in-plot title
+      plot.caption = element_blank(),  # Remove in-plot caption
+      # apply any extra `ggplot2::theme()` args
+      ...
+    )
+  } else {
+    plot <- plot + ggplot2::theme(
+      plot.title = element_blank(),  # Remove in-plot title
+      plot.caption = element_text(hjust = 0),  # Left-align in-plot caption
```

Comment:
This shouldn't be necessary, right? `plot.caption` is already set in `theme_cmap_base()`. If captions should be left-aligned, we should make the change there and allow it to pass through here just by the absence of `plot.caption = element_blank()`. Doing this would allow the user to override the caption justification in `theme_cmap()`, which this implementation does not allow for.

Reply:
I have no issue with that change